src/global/reco/GeneratedJets_factory.cc

```diff
@@ -1,52 +1,72 @@
 // SPDX-License-Identifier: LGPL-3.0-or-later
 // Copyright (C) 2023 Zhongling Ji, Derek Anderson
 
-#include "GeneratedJets_factory.h"
-
+// standard c includes
 #include <memory>
-
 #include <JANA/JEvent.h>
 #include <services/log/Log_service.h>
-
+// event data model definitions
 #include <edm4hep/MCParticle.h>
+// factory-specific includes
+#include "GeneratedJets_factory.h"
+
 
 
 namespace eicrecon {
 
     void GeneratedJets_factory::Init() {
 
-        // This prefix will be used for parameters
-        std::string param_prefix = "reco:" + GetTag();
-
-        // Set input data tags properly
-        InitDataTags(param_prefix);
-
         // SpdlogMixin logger initialization, sets m_log
-        InitLogger(param_prefix, "info");
+        InitLogger(GetPrefix(), "info");
 
+        // initialize jet reconstruction algorithm
         m_jet_algo.init(m_log);
-    }
 
-    void GeneratedJets_factory::ChangeRun(const std::shared_ptr<const JEvent> &event) {
-        // Nothing to do here
-    }
+    }  // end 'Init()'
+
+
+
+    void GeneratedJets_factory::BeginRun(const std::shared_ptr<const JEvent> &event) {
+
+      // nothing to do here
+
+    }  // end 'BeginRun(std::shared_ptr<JEvent&>)'
+
+
 
     void GeneratedJets_factory::Process(const std::shared_ptr<const JEvent> &event) {
-        auto mc_particles = event->Get<edm4hep::MCParticle>("MCParticles");
-
-        std::vector<const edm4hep::LorentzVectorE*> momenta;
-        for (const auto& p : mc_particles) {
-          if (p -> getGeneratorStatus() == 1) {
-            const auto& mom    = p -> getMomentum();
-            const auto& energy = p -> getEnergy();
-            momenta.push_back(new edm4hep::LorentzVectorE(mom.x, mom.y, mom.z, energy));
+
+        // loop over input tags
+        size_t iInput = 0;
+        for (const std::string& input_tag : GetInputTags()) {
```

Comment:
Why are you iterating over input collections? This can't work because down below you overwrite previous result with the latest collection.

I would stick to just one input particle collection `GetInputTags()[0]` and leave the other inputs reserved for possible future use. If we need to merge multiple particle collections, let's make that code into a separate factory.

Reply:
Ah! This is a holdover from my misunderstanding of what a multifactory was for: my thinking was that I would loop over specified input collections and produce an output for each one. I didn't remove it in the last batch of commits just because I thought it wouldn't make a difference.

But I agree with your suggestion: I'll stick to `GetTags()[0]` and reserve the others for future use!